src/shared_utilities/FileReader.jl

```diff
@@ -0,0 +1,413 @@
+
+This module coordinates reading of input data from NetCDF files,
+and allows us to read in data containing a time component.
+It also includes regridding and temporal interpolations
+of this data.
+
+This is based on ClimaCoupler.jl's BCReader and TimeManager modules.
+"""
+module FileReader
+
+using ClimaComms
+using ClimaCore: Fields, Spaces
+using Dates
+using JLD2
+import CFTime
+
+using ClimaLSM.Regridder
+
+export to_datetime,
+    datetime_to_strdate,
+    FileInfo,
+    FileState,
+    SimInfo,
+    PrescribedData,
+    prescribed_data_init,
+    read_data_fields!,
+    next_date_in_file,
+    interpolate_data
+
+
+"""
+    FileInfo
+
+Stores information about the current data being read in from a file.
+
+# Inputs:
+- regrid_dirpath::String     # directory for storing files used in regridding
+- outfile_root::String       # root for files generated when reading data from input file
+- varname::String            # name of the variable we're reading from the input file
+- all_dates::V               # vector containing all dates of the input file
+- date_idx0::I               # index of the first data in the file being used for this simulation
+"""
+struct FileInfo{V, I}
+    regrid_dirpath::String
+    outfile_root::String
+    varname::String
+    all_dates::V
+    date_idx0::I
+end
+
+"""
+    FileState
+
+Stores information about the current data being read in from a file.
+
+# Inputs:
+- data_fields::F        # tuple of two fields at consecutive dates, that will be used for interpolation
+- date_idx::I           # index of the data in the file currently being used
+- segment_length::I     # length of each date segment (used in the daily interpolation)
+"""
+struct FileState{F, I}
+    data_fields::F
+    date_idx::I
+    segment_length::I
+end
+
+"""
+    SimInfo
+
+Stores information about the simulation being run. We may want to store
+multiple copies of an instance of this struct in multiple PrescribedData
+objects if we're reading in data over time for multiple variables.
+
+# Inputs:
+- date_ref::D   # a reference date before or at the start of the simulation
+- t_start::FT    # time in seconds since `date_ref`
+"""
+struct SimInfo{D, FT}
+    date_ref::D
+    t_start::FT
+end
+
+"""
+    PrescribedData
+
+Stores information specific to each prescribed variable from a file.
+Each of the fields of this struct is itself a struct.
+
+# Inputs:
+- file_info::FI     # unchanging info about the input data file
+- file_state::FS    # info about the dates currently being read from file
+- sim_info::S       # unchanging info about the start date/time of the simulation
+"""
+struct PrescribedData{FI, FS, S}
+    file_info::FI
+    file_state::FS
+    sim_info::S
+end
+
+
+"""
+    prescribed_data_init(
+        regrid_dirpath,
+        input_file,
+        varname,
+        date_ref,
+        t_start,
+        surface_space;
+        mono = true,
+    )
+
+Regrids from lat-lon grid to cgll grid, saving the output in a new file,
```

Comment:
maybe: "Regrids from the input lat-lon grid to the simulation cgll grid, saving the regridded output in a new file found at `regrid_dirpath`, and returns the info required to run the simulation using this prescribed data package into a single struct."
?

Reply:
That sounds good to me :)